tests/test_bot_commands.py

```diff
@@ -0,0 +1,63 @@
+import os
+import socket
+
+import discord
+import discord.ext.commands as commands
+import discord.ext.test as dpytest
+import pytest
+import pytest_asyncio
+
+from scutoid import Scutoid
+from util import read_yaml
+
+ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+
+REAL_NAMES = read_yaml(os.path.join(ROOT_DIR, "../real_names.yaml"))
+
+
+@pytest_asyncio.fixture
+async def bot():
+    # Setup
+    intents = discord.Intents.default()
+    intents.members = True
+    intents.message_content = True
+    nicknamer = commands.Bot(command_prefix="!", intents=intents)
+
+    # noinspection PyProtectedMember
+    await nicknamer._async_setup_hook()
+    await nicknamer.add_cog(Scutoid(nicknamer, REAL_NAMES))
+
+    dpytest.configure(nicknamer)
+
+    yield nicknamer
+
+    # Teardown
+    await dpytest.empty_queue()  # empty the global message queue as test teardown
+
+
+@pytest.mark.asyncio
+async def test_nick(bot):
```

Comment:
is there a teardown for this test (putting back the old name?)

Reply:
Well so on testing it does appear to tear down/reset the server state between each test. This makes sense because, I mean, all of the state is in the fixture, so it all goes out of scope after the test.